core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -578,6 +592,10 @@ private ClientFactoryOptions buildOptions() {
             return ClientFactoryOption.ADDRESS_RESOLVER_GROUP_FACTORY.newValue(addressResolverGroupFactory);
         });
 
+        final ProxyConfigSelector selector = WrappingProxyConfigSelector.of(ProxySelector.getDefault());
+        options.computeIfAbsent(ClientFactoryOption.PROXY_CONFIG_SELECTOR, k ->
+                ClientFactoryOption.PROXY_CONFIG_SELECTOR.newValue(selector));
```

Comment:
`selector` could be created inside this lambda expression to avoid unnecessary instantiation of `WrappingProxyConfigSelector`.